src/test/java/core/basesyntax/ServiceShopForTestsImplTest.java

```diff
@@ -0,0 +1,129 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ServiceShopForTestsException;
+import core.basesyntax.service.operation.ServiceShopForTests;
+import core.basesyntax.service.operation.ServiceShopForTestsImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class ServiceShopForTestsImplTest {
+    private ServiceShopForTests serviceShopForTests = new ServiceShopForTestsImpl();
+
+    @Test
+    void service_FruitWithNullParameters_NotOk() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        fruitTransaction.setAmount(200);
+        fruitTransaction.setFruit(null);
+        assertThrows(ServiceShopForTestsException.class, () ->
+                serviceShopForTests.shopTest(fruitTransaction));
+    }
+
+    @Test
+    void service_OperationWithNullParameters_NotOk() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(null);
+        fruitTransaction.setAmount(200);
+        fruitTransaction.setFruit("banana");
+        assertThrows(ServiceShopForTestsException.class, () ->
+                serviceShopForTests.shopTest(fruitTransaction));
+    }
+
+    @Test
+    void service_OperationParametersContainsOneLetter_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.mapToOperation("b"));
+        fruitTransaction.setAmount(50);
+        fruitTransaction.setFruit("banana");
+        Storage.storage.put(fruitTransaction.getFruit(), fruitTransaction.getAmount());
+        try {
+            serviceShopForTests.shopTest(fruitTransaction);
+        } catch (ServiceShopForTestsException e) {
+            fail("something had wrong parameters! ", e);
+        }
```

Comment:
you should not catch exceptions in test classes, you should either except that data is ok ot expect the exception with assertThrows